### PR TITLE
Implement Fighting Initiate - GWF Feat && Champion Fighter "Additional Fighting Style"

### PR DESCRIPTION
**Pull request by Aeristoka**
Behaves the same as Great Weapon Fighting Style

### CODE REVIEW

**Comment by kakaroto** [comment]
Wondering if the 'hasClassFeature' shouldn't just look for the string (an 'includes' instead of '==') and use "Great Weapon Fighting" instead of "Fighting styel.." so it matches all 3.
I've been thinking of something like that as well for the polearm claws/natural weapons thing as well (for that one, I'd do a 'isNaturalWeapon' probably) as, a way to simplify the long if statements.
What do you think ?

**Comment by Aeristoka** [comment]
> Wondering if the 'hasClassFeature' shouldn't just look for the string (an 'includes' instead of '==') and use "Great Weapon Fighting" instead of "Fighting styel.." so it matches all 3.
> I've been thinking of something like that as well for the polearm claws/natural weapons thing as well (for that one, I'd do a 'isNaturalWeapon' probably) as, a way to simplify the long if statements.
> What do you think ?

Based on how they keep adding things in like this, yeah I think that would be better.

**Comment by Aeristoka** [comment]
> Wondering if the 'hasClassFeature' shouldn't just look for the string (an 'includes' instead of '==') and use "Great Weapon Fighting" instead of "Fighting styel.." so it matches all 3.
> I've been thinking of something like that as well for the polearm claws/natural weapons thing as well (for that one, I'd do a 'isNaturalWeapon' probably) as, a way to simplify the long if statements.
> What do you think ?

Should also be done for the 'hasFeat' I think, to simplify things out. In this example, I bet this drops the ' (UA)' at the end, and as it is now, code has to be changed when that changes. If it's using an 'includes' the code can just sit.

**Comment by kakaroto** [comment]
Is the UA still required here or did this change with Tasha's release ?

**Comment by Aeristoka** [comment]
> Is the UA still required here or did this change with Tasha's release ?

It looks like DnDBeyond has this feat screwed up now... You can select the feat, but CAN'T choose the Fighting Style...

**Comment by Aeristoka** [comment]
> Is the UA still required here or did this change with Tasha's release ?

This is fixed now.

**Comment by Aeristoka** [comment]
> Wondering if the 'hasClassFeature' shouldn't just look for the string (an 'includes' instead of '==') and use "Great Weapon Fighting" instead of "Fighting styel.." so it matches all 3.
> I've been thinking of something like that as well for the polearm claws/natural weapons thing as well (for that one, I'd do a 'isNaturalWeapon' probably) as, a way to simplify the long if statements.
> What do you think ?

`hasClassFeature(name) {
        return this._class_features.includes(name);
    }
    hasRacialTrait(name) {
        return this._racial_traits.includes(name);
    }
    hasFeat(name) {
        return this._feats.includes(name);`

Looks like the relevant code for this, not sure how to get a substring of an array.includes() call...

File: src/dndbeyond/base/character.js [hunk 1]
```diff
@@ -343,17 +343,21 @@ class Character extends CharacterBase {
...
     }
-    hasFeat(name) {
-        return this._feats.includes(name);
+    hasFeat(name, substring=false) {
+        if (substring) return this._feats.some(f => f.includes(name));
+        else return this._feats.includes(name);
     }
     hasAction(name) {
-        return this._actions.includes(name);
+        if (substring) return this.actions.some(f => f.includes(name));
```
Comment (kakaroto) [inline src/dndbeyond/base/character.js:359]: really, dude? really?
Can't even use the right variable name.
Much disappointed.

Comment (Aeristoka) [reply src/dndbeyond/base/character.js]: Fixed, and squashed and everything.

**Pull request merged**